packages/fx-core/src/plugins/resource/aad/utils/common.ts

```diff
@@ -63,4 +64,30 @@ export class Utils {
     const tenantId: string = (tokenObject as any).tid;
     return tenantId;
   }
+
+  public static getAndMoveInput(ctx: PluginContext): boolean {
+    if (!isMultiEnvEnabled()) {
+      const skip = ctx.config.get(ConfigKeys.skip) as boolean;
+      return skip;
+    }
+
+    const objectId = ctx.envInfo.config.auth?.objectId;
+    const clientId = ctx.envInfo.config.auth?.clientId;
+    const oauth2PermissionScopeId = ctx.envInfo.config.auth?.accessAsUserScopeId;
+    const clientSecret = ctx.envInfo.config.auth?.clientSecret;
+    const skip = ctx.envInfo.config.auth?.[ConfigKeys.skip];
+
+    if (objectId && clientId && oauth2PermissionScopeId && clientSecret) {
```

Comment:
Do we need to throw errors when something is undefined?

Reply:
Now check will be performed separately when provisioning or skipping provision. It's a good idea to throw error here. I will update the code. Thx!

Reply:
Failure one information not complete?